### PR TITLE
Replace fast-deep-equal with Node.js built-in

### PR DESCRIPTION
**Pull request by jonahsnider**
Remove the `fast-deep-equal` dependency, replacing it with Node's built-in `util` library.

### CODE REVIEW

**Review by donavanbecker — APPROVED** [review]

**Comment by hjdhjd** [comment]
I would not recommend this change. While functionally identical, these are core tests that are key in ensuring ciao is highly performant. fast-deep-equal is a couple of orders of magnitude faster...and given how often these checks will occur, it's a real performance hit. :smile: Here's just one performance benchmark...but there are lots of others out there: https://blog.alexefimenko.com/posts/compare-objects

Comparing objects is hard stuff because the structures are so funky. We'd almost be better off creating a completely bespoke version for ciao for the use cases we need that might squeeze out a shade more performance...but that's a project for another day.

TL;DR: I would not recommend we do this for performance considerations unless we have data otherwise.
